### DISCUSSION TITLE
API, /hardware Status label type is required

### DISCUSSION THREAD

**Discussion by Lokiarmos** [snipe/snipe-it#10134] [Q&A] [answered]
So i have been trying to import my data into my local instance of Snipe, so far everything has been working successfully but when i try to add an asset i get the following error. 

Invoke-WebRequest : {"status":"error","messages":{"type":["Status label type is required."]},"payload":null}

this is the JSON data set that was created via the API Explorer page which i put in just to verify that my powershell code was creating valid JSON which it does. 
{"archived":false,
"depreciate":false,
"requestable":false,
"last_audit_date":"null",
"location_id":83,
"asset_tag":"10981",
"status_id":1,
"model_id":302,
"serial":"C8631470F9A0",
"name":"VIVI"}

so i have reviewed the status and the status that i am after is ID 1 and is valid. 
This is the output of the API Fetch for StatusLabel ID 1
{"id":1,
"name":"Pending",
"type":"pending",
"color":"#ad7333",
"show_in_nav":false,
"default_label":false,
"assets_count":0,
"notes":"These assets are not yet ready to be deployed,
 usually because of configuration or waiting on parts.",
"created_at":null,
"updated_at":{"datetime":"2021-07-27 15:55:21",
"formatted":"2021-07-27 03:55 PM"},
"available_actions":{"update":true,
"delete":true}}

Snipe-IT version
v5.1.8 build 6227 (g6b7d5ed5a)
Software License
AGPL3
PHP Version
7.4.3
Laravel Version
6.20.29

Any idea what the issue might be, also the exact same issue appears if i leave the field status_id out of the JSON object so maybe the field name is wrong?

**Comment by snipe** [comment]
Can you show me your curl request?

**Reply by Lokiarmos** [reply]
Hi, i am not using curl but powershell for doing the commands. 
$response = Invoke-WebRequest -Uri $Uri -Method POST -Headers $headers -ContentType 'application/json' -Body '{"archived":false,"depreciate":false,"requestable":false,"last_audit_date":"null","location_id":83,"asset_tag":"10981","status_id":1,"model_id":302,"serial":"C8631470F9A0","name":"VIVI"}'

**Comment by Lokiarmos** [comment, marked as answer]
So i figured it out, it is the order which the data is stored in the JSON object, 
i used curl on the Snipe server and tested it, entry at a time and finally got it work with this value
{
    "archived": false,
    "depreciate": false,
    "requestable": false,
    "last_audit_date": "null",
    "asset_tag": "10981-PS",
    "status_id": 1,
    "model_id": 302
}

the interesting thing was that if i put model_id before status_id it would throw an issue, and if model_id was before asset_tag it would throw an error for model_id.  the interesting thing is appear that all 3 need to be in order as they are in the API Documents but you can have archived, depreciate and other fields in front of it but asset_tag, status_id and model_id must go in that exact order, it does not matter where as long as they come one after each other in that order...

I don't know if that's by designed but at least i know now..

**Reply by boring10** [reply]
I wonder if this only applies when you are not using auto incrementing asset tags. I've never run into that and I know for sure that I have has the status_id all over the place for creating new assets.
